### PR TITLE
Refine natural keys for various DCIM models

### PR DESCRIPTION
**Pull request by timizuoebideri1**
# Closes: #3972 
# What's Changed

# TODO

- [ ] Explanation of Change(s)
- [ ] Added change log fragment(s) (for more information see [the documentation](https://docs.nautobot.com/projects/core/en/stable/development/#creating-changelog-fragments))
- [ ] Attached Screenshots, Payload Example
- [ ] Unit, Integration Tests
- [ ] Documentation Updates (when adding/changing features)
- [ ] Example Plugin Updates (when adding/changing features)
- [ ] Outline Remaining Work, Constraints from Design

### CODE REVIEW

File: nautobot/dcim/models/racks.py [hunk 1]
```diff
@@ -477,8 +477,7 @@ class RackReservation(PrimaryModel):
     class Meta:
         ordering = ["created"]
 
-    # TODO: no appropriate natural-key for RackReservation?
-    natural_key_field_names = ["id"]
+    natural_key_field_names = ["units", "rack"]
```
Comment (itdependsnetworks) [inline nautobot/dcim/models/racks.py:480]: I would guess this is better off reverse?

Comment (glennmatthews) [reply nautobot/dcim/models/racks.py:480]: no, it needs to be this way because the natural-key for a `Rack` itself is variable-length.

**Comment by timizuoebideri1** [comment]
Do i still need to write test for these models with changed `natural_key`? 🤷🏽‍♂️
Since there are test already for `natural_key`

File: nautobot/dcim/models/device_components.py [hunk 2]
```diff
@@ -400,6 +402,8 @@ class PowerOutlet(CableTermination, PathEndpoint, ComponentModel):
         help_text="Phase (for three-phase feeds)",
     )
 
+    natural_key_field_names = ["name", "device"]
```
Comment (glennmatthews) [inline nautobot/dcim/models/device_components.py:405]: Shouldn't be needed I think since `PowerOutlet` inherits from `ComponentModel`?

**Review by glennmatthews — APPROVED** [review]

File: nautobot/dcim/models/device_components.py (hunk 2 above)
Comment (timizuoebideri1) [inline nautobot/dcim/models/device_components.py:405]:
```suggestion
```

**Pull request merged**